src/mbedtls.h

```diff
@@ -278,6 +278,12 @@
 
 #if LIBSSH2_ECDSA
 
+/* force expose internal mbedtls function for linking,
+   I expect this won't change internally until mbedtls 4.0 */
+#if MBEDTLS_VERSION_MAJOR == 3 && MBEDTLS_VERSION_MINOR >= 6
+int mbedtls_pk_load_file(const char *path, unsigned char **buf, size_t *n);
+#endif
+
```

Comment:
```suggestion
/* Force-expose internal mbedTLS function */
#if MBEDTLS_VERSION_NUMBER >= 0x03060000
int mbedtls_pk_load_file(const char *path, unsigned char **buf, size_t *n);
#endif

```

Also move this block into `mbedtls.c` before the function where we are using this.